### PR TITLE
Record AVX use for SIMD indirections

### LINKED ISSUE #21055: 32-bit SIMD indirections don't flag use of AVX
The `LinearScan::Build` methods are intended to call `SetContainsAVXFlags` for any potential use of AVX registers. However, this is not done in LinearScan::BuildIndir()`, which can encounter loads or stores of 32-bit SIMD types (and note that the general case in `LinearScan::BuildNode()`, which calls this method for any tree for which `varTypeIsFloating` is true, doesn't cover this case because that only covers `TYP_FLOAT` and `TYP_DOUBLE`).

### PR DESCRIPTION
**Pull request by CarolEidt**
Fix #21055

### CODE REVIEW

**Comment by CarolEidt** [comment]
@fiigii @tannergooding @dotnet/jit-contrib PTAL

**Comment by CarolEidt** [comment]
PMI Diffs for frameworks & benchmarks:
x64: Total bytes of diff: 13873 (0.03% of base)
x86: Total bytes of diff: 10201 (0.03% of base)

**Comment by CarolEidt** [comment]
> Is it a regression?

Yes, it is a regression, as there are places where we were not inserting the vzeroupper before, where we should have been (at least according to the current "rules"), and those are now being added.

File: src/jit/lsraxarch.cpp [hunk 1]
```diff
@@ -2824,6 +2824,10 @@ int LinearScan::BuildIndir(GenTreeIndir* indirTree)
         }
     }
 #ifdef FEATURE_SIMD
+    if (varTypeIsSIMD(indirTree))
+    {
+        SetContainsAVXFlags(true, genTypeSize(indirTree->TypeGet()));
```
Comment (tannergooding) [inline src/jit/lsraxarch.cpp]: I wonder if there is an assert we could do in the emitter to catch these whenever we emit a relevant instruction....

**Pull request merged**